staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -139,15 +143,24 @@ func (h *kmsPluginProbe) Check() error {
 		return h.lastResponse.err
 	}
 
-	p, err := h.Service.Encrypt([]byte("ping"))
-	if err != nil {
-		h.lastResponse = &kmsPluginHealthzResponse{err: err, received: time.Now()}
-		return fmt.Errorf("failed to perform encrypt section of the healthz check for KMS Provider %s, error: %v", h.name, err)
-	}
+	if err := h.Service.Check(); err != nil {
```

Comment:
Why does the "Service" need to know about the "Check"?
We have all we need here to call the gRPC's health service. This approach would also allow us to use different time-out and other connection's settings for the healthz.

Reply:
we are going to re-use the NewGRPCService function.